### PR TITLE
Authentication on the backend

### PR DESCRIPTION
**Pull request by Dominik-Mika**

### CODE REVIEW

File: src/main/java/eu/elision/pricing/domain/User.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+import java.util.UUID;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+
+@Getter
```
Comment (Gschrooyen) [inline src/main/java/eu/elision/pricing/domain/User.java:21]: use @Data

File: src/main/java/eu/elision/pricing/security/SecurityConfig.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            .sessionCreationPolicy(SessionCreationPolicy.STATELESS)
+            .and().authenticationProvider(authenticationProvider)
+            .addFilterBefore(jwtAuthenticationFilter, UsernamePasswordAuthenticationFilter.class);
+
+        return http.build();
+    }
+
+    @Bean
+    public AuthenticationProvider authenticationProvider(UserDetailsService userDetailsService) {
+        DaoAuthenticationProvider daoAuthenticationProvider = new DaoAuthenticationProvider();
```
Comment (Gschrooyen) [inline src/main/java/eu/elision/pricing/security/SecurityConfig.java:47]: autowire this

**Review by Gschrooyen — CHANGES_REQUESTED** [review]
add javadoc everywhere

**Review by rodzers-usackis — APPROVED** [review]

**Review by rodzers-usackis — APPROVED** [review]

**Comment by Dominik-Mika** [comment]
Requested changes implemented.

**Pull request merged**